### ISSUE TITLE
♻️ Remove deprecated `hasDiploma` field 

### ISSUE THREAD

**Issue by vd1992** [GCTC-NTGC/gc-digital-talent#8661]
## ♻️ Debt/Refactor

The `hasDiploma` field has been marked as deprecated for sometime and should be removed. 

## 🕵️ Details

The field exists and is referenced in various places, some places where it explicitly is set to do nothing like in filtering.
It ought to be removed

## 🙋‍♀️ Proposed Solution

Remove it
- from schema
- from PHP unit tests
- from snapshot generation
- from any front end instances, like the filters where it is disconnected from API
- Database
- Use caution

## ✅ Acceptance Criteria

- [ ] Removed from where decided in Proposed solution

**Comment by brindasasi** [comment]
I just see that the filter exists in Pool Candidates table filter in UAT. But it doesn't filter anything based on its selected value. 
![image (27)](https://github.com/GCTC-NTGC/gc-digital-talent/assets/89482910/f5330da5-d67b-4b30-b924-c933560926d3)

**Comment by vd1992** [comment]
Yep, it is present in multiple places but disconnected from API making it a filter that does nothing

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/521a3c78-52d0-467a-ba09-83b1ccc43f1c)

**Comment by petertgiles** [comment]
This issue got tricky because of the larger question about how education is handled.  n https://github.com/GCTC-NTGC/gc-digital-talent/issues/8821

**Comment by mnigh** [comment]
@tristan-orourke Is `hasDiploma` still deprecated and if so, can it finally be removed? If it's not deprecated, we should probably remove the code that deprecated it in the codebase.

**Comment by mnigh** [comment]
@petertgiles same question as https://github.com/GCTC-NTGC/gc-digital-talent/issues/8661#issuecomment-2605249804.

**Comment by petertgiles** [comment]
I haven't heard anything new about education lately.  No decisions have been made, I think.

**Comment by mnigh** [comment]
> I haven't heard anything new about education lately. No decisions have been made, I think.

Ok, I've created https://github.com/GCTC-NTGC/gc-digital-talent/issues/12649.

**Comment by petertgiles** [comment]
Going with with removing deprecation notice for now
